File: Array/_Container with most water.cs
public class Solution {
    public int MaxArea(int[] height) {
        int max = 0;
        int left=0, right=height.Length-1;

        while (left<right){
            int value = (right-left)*Math.Min(height[left],height[right]);
            if (max<value){
                max=value;
            }

            // when left and right go inward of the array
            // we want to keep the edge height as high as possible
            // so we compare two heights and move the shortest one inward first 
            if(height[left]<=height[right]){
                left++;
            }
            else{
                right--;
            }
        }
    return max;
    }
}

File: Array/Container With Most Water.py
class Solution(object):
    def maxArea(self, height):
        left,right=0,len(height)-1
        res=0
        while left<right:
            a=min(height[left],height[right])*(right-left)
            if a>res:res=a
            if height[left]<height[right]:
                left+=1
            else:right-=1
        return res